ansible/rke2/airgap/inventory/group_vars/all.yml.template

```diff
@@ -0,0 +1,92 @@
+---
+# Global variables for RKE2 airgap deployment with tarball installation
+
+# RKE2 Configuration
+rke2_version: "v1.31.11+rke2r1"
+rke2_server_options: |
+  cluster-cidr: {{ cluster_cidr }}
+  service-cidr: {{ service_cidr }}
+  cluster-dns: {{ cluster_dns }}
+  disable:
+    - rke2-snapshot-controller
+    - rke2-snapshot-controller-crd
+    - rke2-snapshot-validation-webhook
+rke2_agent_options: ""
+
+# Installation Method Configuration
+# Available methods:
+#   - "tarball": Uses pre-downloaded RKE2 tarballs with embedded images (no registry required)
+#     Playbook: rke2-tarball-playbook.yml
+#     Best for: Pure airgap environments, simple deployments
+installation_method: "tarball"
+
+# SSH Configuration
+# This only needs to be set here or in the inventory file, but not both
+ssh_private_key_file: "~/.ssh/id_rsa"
+
+# Network Configuration
+cluster_cidr: "10.42.0.0/16"
+service_cidr: "10.43.0.0/16"
+cluster_dns: "10.43.0.10"
+
+# CNI Configuration
+# Available CNI options:
+#   - "canal" (default): Flannel + Calico for network policy (RKE2 default)
+#   - "calico": Calico CNI for advanced networking and network policies
+#   - "cilium": Cilium CNI for advanced networking, security, and observability
+#   - "multus": Enable Multus for multiple network interfaces (can be combined with others)
+#   - "none": Disable CNI installation (bring your own CNI)
+cni_plugin: "canal"  # Default RKE2 CNI
+
+# CNI-specific configuration
+cni_config:
+  # Calico specific settings
+  calico:
+    # IP-in-IP encapsulation mode: Always, CrossSubnet, Never
+    ipip_mode: "Always"
+    # BGP configuration
+    bgp_enabled: true
+    # Network policy enforcement
+    network_policy: true
+    
+  # Cilium specific settings
+  cilium:
+    # Enable Hubble for observability
+    hubble_enabled: true
+    # Enable network policies
+    network_policy: true
+    # Enable BGP
+    bgp_enabled: false
+    # Enable service mesh features
+    service_mesh: false
+    
+  # Canal specific settings (Flannel + Calico)
+  canal:
+    # Flannel backend: vxlan, host-gw, udp
+    flannel_backend: "vxlan"
+    # Enable Calico network policies
+    network_policy: true
+    
+  # Multus specific settings
+  multus:
+    # Default CNI to use with Multus (canal, calico, cilium)
+    default_cni: "canal"
```

Comment:
Is this required? I feel like the other option should just pass into the rke2 config.yaml directly

Reply:
I'd be inclined to refactor the template to just take in a config.yaml instead of having the ability to adjust CNI options directly here, but it's fine as is and we can refactor as a separate effort if/when needed

Reply:
yeah let's add that later, if it's needed